Docs/add asemmbly introduction (#415)
iota-community/iota-wiki: internal/learn/future/assembly.md

---
id: assembly
title: Assembly
description: Introduction to the emerging permissionless Smart Contracts network
keywords:
  - smart contracts
  - permissionless
  - ASMB
---

![Assembly](/img/learn/future/assembly.png 'Click to see the full-size image.')

In December 2021, IOTA Foundation [introduced](https://blog.assembly.sc/announcing-assembly-and-the-asmb-token/) Assembly, a permissionless Smart Contracts network. Assembly will function as an intermediate layer for permissionless and interoperable Smart Contracts, and will have its own token called [ASMB](https://assembly.sc/token).
For more information, check the official [Website](https://assembly.sc) or the [Assembly Wiki](https://wiki.assembly.sc).

> # The role of the IOTA Token

Reviewer: ```suggestion
## The role of the IOTA Token
```

Reviewer: We should only have one h1 headline per page (SEO and consistency)

Assembly is build on top of the Tangle, it will need the Tangle for anchored state updates, interoperability between chains, for Native Assets and general access to a congested network. There are two main usages of the IOTA Token in Assembly.

> ## The new dust protection

Reviewer: ```suggestion
### The new dust protection
```

> With the new [dust protection](https://github.com/iotaledger/tips/pull/39), every data stored on the Tangle needs to be backed by IOTA Tokens. So every NFT, SC state update, Native Asset requires IOTA Tokens so that it can be stored on L1. You can read more about the new dust protection in this [community blog post](https://medium.com/@wernerderchamp/dust-protection-on-the-iota-network-an-eli12-d8ca567a2d36).

Reviewer 2: ```suggestion
With the new [dust protection](https://github.com/iotaledger/tips/pull/39), all data stored on the Tangle needs to be backed by IOTA Tokens. For example, every NFT, SC chain state anchor or Native Asset requires a deposit of IOTA Tokens in order to be stored on L1. You can read more about the new dust protection in this [community blog post](https://medium.com/@wernerderchamp/dust-protection-on-the-iota-network-an-eli12-d8ca567a2d36).
```

> ## Mana

Reviewer: ```suggestion
### Mana
```

> Assembly nodes have to anchor the chain state often, even if the Tangle is congested. To get a fair share of the Tangle throughput in the [next version](https://v2.iota.org) of the IOTA protocol, you will need [Mana](/IOTA-2.0-Research-Specifications/5.3Mana), and Mana is derived directly from the IOTA Token. Assembly nodes will have to hold IOTA tokens, which will create even more demand.

Reviewer 2: ```suggestion
Assembly nodes have to anchor the chain state often, even if the Tangle is congested. To get a fair share of the Tangle throughput in the [next version](https://v2.iota.org) of the IOTA protocol, you will need [Mana](/IOTA-2.0-Research-Specifications/5.3Mana), and Mana is derived directly from the IOTA Token. Assembly nodes will have to hold IOTA tokens, which will increase demand.
```

Author: The idea was to format it this way to wrap it up why both the dust thingy and mana increase the demand for IOTA.
Wdyt?

Reviewer 2: So maybe "increase demand even more."? I just feel like "create even more demand" isn't proper wording, something is created only once, so saying something is "created even more" doesn't sound correct. Demand can be decreased or increased though.

Author: ```suggestion
Assembly nodes have to anchor the chain state often, even if the Tangle is congested. To get a fair share of the Tangle throughput in the [next version](https://v2.iota.org) of the IOTA protocol, you will need [Mana](/IOTA-2.0-Research-Specifications/5.3Mana), and Mana is derived directly from the IOTA Token. Assembly nodes will have to hold IOTA tokens, which will increase demand even more.
```

---
id: assembly
title: Assembly
description: Introduction to the emerging permissionless Smart Contracts network
keywords:
  - smart contracts
  - permissionless
  - ASMB
---

![Assembly](/img/learn/future/assembly.png 'Click to see the full-size image.')

In December 2021, IOTA Foundation [introduced](https://blog.assembly.sc/announcing-assembly-and-the-asmb-token/) Assembly, a permissionless Smart Contracts network. Assembly will function as an intermediate layer for permissionless and interoperable Smart Contracts, and will have its own token called [ASMB](https://assembly.sc/token).
For more information, check the official [Website](https://assembly.sc) or the [Assembly Wiki](https://wiki.assembly.com).

# The role of the IOTA Token

> So you maybe asked yourself what the role of IOTA is in this Assembly network. Assembly is build on top of the Tangle, it will need the Tangle for anchored state updates, interoperability between chains, for Native Assets and general access to a congested network. So there are basically two main usages of the IOTA Token in Assembly.

Reviewer 3: ```suggestion
Assembly is build on top of the Tangle, it will need the Tangle for anchored state updates, interoperability between chains, for Native Assets and general access to a congested network. There are two main usages of the IOTA Token in Assembly.
```

## The new dust protection

> With the new [dust protection](https://github.com/muXxer/protocol-rfcs/blob/master/text/0032-dust-protection/0032-dust-protection.md), every data stored on the Tangle needs to be backed by IOTA Tokens. So every NFT, SC state update, Native Asset requires IOTA Tokens so that it can be stored on L1. You can also read more about the new dust protection in this [community blog post](https://medium.com/@wernerderchamp/dust-protection-on-the-iota-network-an-eli12-d8ca567a2d36).

Reviewer 3: ```suggestion
With the new [dust protection](https://github.com/muXxer/protocol-rfcs/blob/master/text/0032-dust-protection/0032-dust-protection.md), every data stored on the Tangle needs to be backed by IOTA Tokens. So every NFT, SC state update, Native Asset requires IOTA Tokens so that it can be stored on L1. You can read more about the new dust protection in this [community blog post](https://medium.com/@wernerderchamp/dust-protection-on-the-iota-network-an-eli12-d8ca567a2d36).
```

> ## Mana

Reviewer 3: What if we rearrange this paragraph?

1. Assembly nodes want to anchor their state even if Tangle is congested
2. Right now Coordinator manages traffic, but after coordicide you will need Mana (link to Mana)
3. To generate mana, you need to own IOTA tokens
4. It means that ASMB does not compete with MIOTA, but creates demand for it.

Author: You mean rearrange only this Mana paragraph?
I just ask because point 4 is the general conclusion of Mana and the new dust protection, right? So should that be it‘s own „paragraph“?

Reviewer 3: Good point. You could finish paragraph with something like ", creating even more demand for the IOTA token". So it would include the dust protection and wrap up the whole article.

Author: Another question for point 2. The coordinator manages traffic doesn‘t sound right to me. I mean yeah indirectly if the coo crashes he controls troughput :trollface: , but mainly mainnet manages traffic by crashing weak nodes right now 😅 
So maybe it‘s more like: `Currently there is no congestion mechanism in place but after coordicide you will need Mana to get your fair share in a congested network`?

Reviewer 3: Ye. Maybe  don't even mention the current state and just go with "after coordicide/in IOTA 2.0 network nodes will need Mana..."

Author: What about now? 😅

> Assembly nodes want to anchor their state even if the Tangle is congested. After coordicide nodes will need Mana to get their fair share of the Tangle throughput. Mana is directly derived from the IOTA Token. 
> So the IOTA Nodes which anchor the Smart Contract state on L1 will need to have Mana (and therefore IOTA tokens) if the network is congested, creating even more demand for the IOTA token.

Reviewer 3: ```suggestion
Assembly nodes have to anchor the chain state often, even if the Tangle is congested. To get a fair share of the Tangle throughput in the [next version](LINK_TO_IOTA_2.0) of the IOTA protocol, you will need [Mana](LINK_TO_MANA), and mana is directly derived from the IOTA Token.  Assembly nodes will have to hold IOTA tokens, which will create even more demand.
```

Diff looks imposing, but I just changed places for a few clauses and had cut a sentence. I think this paragraph is still mostly yours.

Author: Done

---
id: assembly
title: Assembly
description: Introduction to the emerging permissionless Smart Contracts network
keywords:
  - smart contracts
  - permissionless
  - ASMB
---

![Assembly](/img/learn/future/assembly.png 'Click to see the full-size image.')

> In December 2021, IOTA Foundation [introduced](https://blog.assembly.sc/announcing-assembly-and-the-asmb-token/) Assembly, a permissionless Smart Contracts network. Assembly will function as an intermediate layer for permissionless and interoperable Smart Contracts. It will also have its own token called [ASMB](https://assembly.sc/token). The Token will help secure the network.

Reviewer 3: ```suggestion
In December 2021, IOTA Foundation [introduced](https://blog.assembly.sc/announcing-assembly-and-the-asmb-token/) Assembly, a permissionless Smart Contracts network. Assembly will function as an intermediate layer for permissionless and interoperable Smart Contracts, and will have its own token called [ASMB](https://assembly.sc/token).
```

There is too much information for a single paragraph, that's my lazy attempt to fix it.

> This introduction should only give you a brief overview of Assembly. For more information, check the official [Website](https://assembly.sc) or the [Assembly Wiki](https://wiki.assembly.com).

Reviewer 3: Was it intended to be a separate paragraph? (extra line break is missing)

Author: Nope. Do you think it should be?

> ## What is Assembly?

Reviewer 3: Maybe just cut everything from there and on? Looks like doing the same job twice (here and in Assembly Wiki).

Author: Not sure about that. I can drop `What is Assembly`. But it was really important to me to explain the value of IOTA in Assembly as many seem to not understand that and imo it's the most important part to understand the future of IOTA. 🤔

Reviewer 3: I think this article will be better if you focus on why Assembly and Tangle benefit on each other, yes. "Assembly needs Tangle to communicate and secure chains, Tangle benefits from Assembly because chains need mana to commit state and the best way to always have Mana is to run a tangle node".

Author: So you would ony add this single paragraph instead of everything under ``What is Assembly?`?
I'm fine with that but then we need this conntent in the assembly wiki. If it isn't already and I just didn't find it yet ;)
Wdyt?

Reviewer 3: Not necessary all in a single paragraph, but I think you can cut explaining what Assembly is (beyond saying that's a new network), and focus on why Assembly empowers IOTA and not overshadows it, since this is the real question you want to answer.

Author: What do you think about the changes now 😅 ?

Assembly is basically a permissionless ISCP network build on top of IOTA. It fully leverages and depends on the features of its IOTA base layer. All Smart Contract chains can directly communicate with each other over Assembly and also share Native Assets over the IOTA network.

## Permissionless Smart Contracts

> Assembly is IOTAs approach to permissionless Smart Contracts. For that, the root chain got introduced. The root chain is basically the backbone of Assembly. It slashes staked funds, manages and distributes Validator to the different chains. You as a chain creator can choose your validators based on different criteria. Those validators can then join your chain if they want-

Reviewer: ```suggestion
Assembly is IOTAs approach to permissionless Smart Contracts. For that, the root chain got introduced. The root chain is basically the backbone of Assembly. It slashes staked funds, manages and distributes Validator to the different chains. You as a chain creator can choose your validators based on different criteria. Those validators can then join your chain if they want to.
```

## The role of the IOTA Token

So you maybe asked yourself what the role of IOTA is in this Assembly network. As already briefly explained above, Assembly is build on top of IOTA. It will need the IOTA Token for interoperability between chains, for Native Assets and general access to a congested network. So there are basically two main usages of the IOTA Token in Assembly.

### The new dust protection

> With the new [dust protection](https://github.com/muXxer/protocol-rfcs/blob/master/text/0032-dust-protection/0032-dust-protection.md), every data stored on the Tangle needs to be backed by IOTA Tokens. So every NFT, SC state update, Native Asset requires IOTA Tokens so that it can be stored on L1. You can also read more about the new dust protection in this [community blog post](https://github.com/muXxer/protocol-rfcs/blob/master/text/0032-dust-protection/0032-dust-protection.md).

Reviewer: Wrong link for the blog post?

Author: ```suggestion
With the new [dust protection](https://github.com/muXxer/protocol-rfcs/blob/master/text/0032-dust-protection/0032-dust-protection.md), every data stored on the Tangle needs to be backed by IOTA Tokens. So every NFT, SC state update, Native Asset requires IOTA Tokens so that it can be stored on L1. You can also read more about the new dust protection in this [community blog post](https://medium.com/@wernerderchamp/dust-protection-on-the-iota-network-an-eli12-d8ca567a2d36).
```

Author: Thanks :)
